tests/csrc/unittests/test_sampling_layer.cu

```diff
@@ -166,11 +168,16 @@ protected:
         d_output_log_probs = reinterpret_cast<float*>(allocator->malloc(sizeof(float) * max_output_len * batchxbeam));
         d_output_ids       = reinterpret_cast<int*>(allocator->malloc(sizeof(int) * max_seq_len * batchxbeam));
         d_end_ids          = reinterpret_cast<int*>(allocator->malloc(sizeof(int) * batchxbeam));
+        d_curand_state     = reinterpret_cast<curandState_t*>(allocator->malloc(sizeof(curandState_t) * batch_size));
+        d_random_seed =
+            reinterpret_cast<unsigned long long*>(allocator->malloc(sizeof(unsigned long long) * batch_size));
 
         // Init by zero.
         cudaMemset(d_cum_log_probs, 0, sizeof(float) * batchxbeam);
         cudaMemset(d_output_log_probs, 0, sizeof(float) * max_output_len * batchxbeam);
         cudaMemset(d_output_ids, 0, sizeof(int) * max_seq_len * batchxbeam);
+        cudaMemset(d_random_seed, 0, sizeof(unsigned long long) * batch_size);
+        invokeCurandBatchInitialize(d_curand_state, batch_size, d_random_seed, stream);
```

Comment:
I think we should copy `random_seed` to `d_random_seed` and use `d_random_seed` to initialize `d_curand_state`.

Reply:
> I think we should copy `random_seed` to `d_random_seed` and use `d_random_seed` to initialize `d_curand_state`.

May I need to add `h_random_seed`? Currently, it is directly allocated and initialized on the device without a copy process.

Reply:
I mean this argument [random_seed](https://github.com/InternLM/lmdeploy/blob/9feafc1ecec1b9ba9ee23c5bad3e05ebd01a79e8/tests/csrc/unittests/test_sampling_layer.cu#L1048-L1051).  Since it could be size of 1, you should handle this case as the size of d_curand_state is batch_size

Reply:
> I mean this argument [random_seed](https://github.com/InternLM/lmdeploy/blob/9feafc1ecec1b9ba9ee23c5bad3e05ebd01a79e8/tests/csrc/unittests/test_sampling_layer.cu#L1048-L1051). Since it could be size of 1, you should handle this case as the size of d_curand_state is batch_size

make sense ok

Reply:
> Since it could be size of 1, you should handle this case as the size of d_curand_state is batch_size

You are right. When size is 1, two unit tests failed. I'll take a look at what the root cause is.

Reply:
> Since it could be size of 1, you should handle this case as the size of d_curand_state is batch_size

@irexyc done https://github.com/InternLM/lmdeploy/pull/1874/commits/b73f31b9c41a60e77121323213d38f9cfdf88d7b